querybook/server/models/query_review.py

```diff
@@ -0,0 +1,109 @@
+import sqlalchemy as sql
+from sqlalchemy.orm import relationship, backref
+from enum import Enum
+
+from app import db
+from const.db import now, description_length
+from lib.sqlalchemy import CRUDMixin
+from lib.utils.serialize import with_formatted_date
+
+Base = db.Base
+
+
+class QueryReviewStatus(Enum):
+    PENDING = "pending"
+    APPROVED = "approved"
+    REJECTED = "rejected"
+
+
+class QueryReview(Base, CRUDMixin):
+    __tablename__ = "query_review"
+    __table_args__ = {"mysql_engine": "InnoDB", "mysql_charset": "utf8mb4"}
+
+    id = sql.Column(sql.Integer, primary_key=True, autoincrement=True)
+    query_execution_id = sql.Column(
+        sql.Integer,
+        sql.ForeignKey("query_execution.id", ondelete="CASCADE"),
+        nullable=False,
+        unique=True,
+    )
+    query_author_id = sql.Column(
+        sql.Integer, sql.ForeignKey("user.id", ondelete="CASCADE"), nullable=False
+    )
+
+    status = sql.Column(
+        sql.Enum(QueryReviewStatus), nullable=False, default=QueryReviewStatus.PENDING
+    )
+    review_request_reason = sql.Column(
+        sql.String(length=description_length), nullable=True
+    )
+    rejection_reason = sql.Column(sql.String(length=description_length), nullable=True)
```

Comment:
I think request and rejection reason should be both required

Reply:
I realized we will only have rejection reason after reviewer rejects, so will make rejection_reason  nullable=True for  review request creation